text/0003-blockspace-regions.md

```diff
@@ -0,0 +1,252 @@
+```
+
+This API is intended to be wrapped by logic implementing interfaces such as RFC-5.
+
+Created regions for a core MUST NOT exceed a combined block rate of 1 at any block.
+
+### Changes to backing/availability
+
+Regions are used to modify the behavior of the parachain backing/availability pipeline. The first major change is that parachain candidates submitted to the relay-chain in the `ParasInherent` will be annotated with the `RegionId` that they are intended to occupy. Validators and collators do the work of figuring out which blocks are assigned to which regions, lifting the burden of granular scheduling off of the relay chain.
+
+The **maximum consumption** of a region at any block number `now` is given by:
+`maximum_consumption(now, region) = min(now, end) - start * rate`
+If `end` is `None`, it is treated as infinite.
+
+The **minimum consumption** of a region at any block number `now` is given by:
+`minimum_consumption(now, region) = maximum_consumption(now - SCHEDULING_LENIENCE, region)`
+
+The **effective consumption** of a region at any block number `now` is the given by:
+`max(region.consumption, minimum_consumption(now, region))`
+
+A submitted candidate which uses `p: PartsOf57600` of the core's resources for a parachain P at a block B is accepted if:
+  * There is no candidate pending availability for B
+  * `effective_consumption(B, region) + p <= maximum` if `maximum` is `Some`
+  * `effective_consumption(B, region) + p <= maximum_consumption(B, region)`
+
+How core resource limits are defined is left beyond the scope of this RFC - at the time of writing, all cores have the same resource limits, but this design allows cores to be specialized in their resource limits, with some cores allowing more data, some allowing more granularity or execution time, etc.
+
+If all of these conditions are met, along with other validity conditions for backed candidates beyond the scope of this RFC, then the candidate is pending availability and the region's consumption value is incremented to `effective_consumption(B, region) + p` If the candidate times out before becoming available, the consumption value is reduced by `p`.
+
+The scheduling lenience allows regions to fall behind their expected tickrate, limited to a small maximum level of debt. This prevents accumulated core debt from being accumulated indefinitely and spent when convenient. Smoothing system load over short time horizons is desirable, but over infinite time horizons becomes dangerous.
+
+This RFC introduces a new `HYPERCORES` parameter into the `HostConfiguration` which relay-chain governance uses to manage the parameters of the parachains protocol. Hypercores are inspired by technologies such as hyperthreading, to emulate multiple logical cores on a single phsyical core as resources permit. Hypercores allow parachains to make up for missed scheduling opportunities, which is important to effectively decouple parachain growth from backing on the relay chain.
```

Comment:
Why `HYPERCORES`  instead of simply some separate `burstable` or `reserved` cores? 
Parachains can only be assigned such a core when they need to make up for missed scheduling opportunities.  
I think it to be a simpler way of solving (3). 

Also I am not a fan of `hyperthreading`, it's a something known for it's security flaws and currently the recommendation is to disable it.

WDYM by `decouple parachain growth from backing on the relay chain`, AFAIK all parachain blocks should be backed.

Reply:
I'm OK with calling them "burst" cores or something like it.

> WDYM by decouple parachain growth from backing on the relay chain, AFAIK all parachain blocks should be backed.

Specifically it refers to avoiding scheduling friction between multiple chains sharing the same core. Burst cores are a way to prevent explicit friction by trading off temporary system performance. This is much better than trying to account for that friction, which would need to cascade all the way into validators' and collators' logic. i.e. we decouple the growth of independent parachains sharing the same core.